std/math/emulated/field_ops.go

```diff
@@ -132,6 +133,34 @@ func (f *Field[T]) add(a, b *Element[T], nextOverflow uint) *Element[T] {
 	return f.newInternalElement(limbs, nextOverflow)
 }
 
+func (f *Field[T]) AddMany(inputs ...*Element[T]) *Element[T] {
+	if len(inputs) == 0 {
+		return f.Zero()
+	}
+	overflow := uint(0)
+	nbLimbs := 0
+	for i := range inputs {
+		f.enforceWidthConditional(inputs[i])
+		if inputs[i].overflow > overflow {
+			overflow = inputs[i].overflow
+		}
+		if len(inputs[i].Limbs) > nbLimbs {
+			nbLimbs = len(inputs[i].Limbs)
+		}
+	}
+	addOverflow := bits.Len(uint(len(inputs)))
+	limbs := make([]frontend.Variable, nbLimbs)
+	for i := range limbs {
+		limbs[i] = 0
+	}
+	for i := range inputs {
+		for j := range inputs[i].Limbs {
+			limbs[j] = f.api.Add(limbs[j], inputs[i].Limbs[j])
+		}
+	}
+	return f.newInternalElement(limbs, overflow+uint(addOverflow))
```

Comment:
so, the final overflow is the max input overflow + `addOverflow` which is .. `bits.Len(nbInput)` ? 
if nbInput == 1 --> `addOverflow` should be 0 right? 
why is the `addOverflow` not `nbInputs - 1` ?

Reply:
So, in case when we add two elements, we increase the overflow by one to account for additional bit required.

Now, when we add many elements, then the bound on the sum is `nbInputs * 2^(maxOverflow + nbBitsPerLimb)`. Its length is `log2(nbInputs * 2^(maxOverflow + nbBitsPerLimb)) = log2(nbInputs) + maxOverflow + nbBitsPerLimb`. When counting only the overflow (so can omit `nbBitsPerLimb` part), then it is `log2(nbInputs) + maxOverflow` and we need to round `log2(nbInputs)` up.